File: app/src/main/java/com/simplemobiletools/calculator/operation/PowerOperation.java
package com.simplemobiletools.calculator.operation;

import com.simplemobiletools.calculator.operation.base.BinaryOperation;
import com.simplemobiletools.calculator.operation.base.Operation;
import com.simplemobiletools.calculator.operation.base.UnaryOperation;

public class PowerOperation extends BinaryOperation implements Operation {

    protected PowerOperation(double baseValue, double secondValue) {
        super(baseValue, secondValue);
    }

    @Override
    public double getResult() {
        double result = Math.pow(baseValue, secondValue);
        if (Double.isInfinite(result) || Double.isNaN(result))
            result = 0;
        return result;
    }
}

File: app/src/main/kotlin/com/simplemobiletools/calculator/operation/PowerOperation.kt
package com.simplemobiletools.calculator.operation

import com.simplemobiletools.calculator.operation.base.BinaryOperation
import com.simplemobiletools.calculator.operation.base.Operation
import kotlin.math.pow

class PowerOperation(baseValue: Double, secondValue: Double) : BinaryOperation(baseValue, secondValue), Operation {

    override fun getResult(): Double {
        val result = baseValue.pow(secondValue)

        return if (java.lang.Double.isInfinite(result) || java.lang.Double.isNaN(result))
            0.0
        else {
            result
        }
    }
}
